scheduler/test/cook/test/mesos/util.clj

```diff
@@ -562,4 +563,82 @@
     (is (= 12 (util/cache-lookup! cache-store "A" 12)))
     (is (= {"A" 12, "C" 31} (.cache @cache-store)))))
 
+
+(deftest test-retry-job
+  (let [uri "datomic:mem://test-retry-job"
+        conn (restore-fresh-database! uri)]
+    (testing "increment retries on running job"
+      (let [[job _] (create-dummy-job-with-instances conn :job-state :job.state/running
+                                                     :instances [{:instance-status :instance.status/running}])
+            uuid (:job/uuid (d/entity (d/db conn) job))]
+        (util/retry-job! conn uuid 10)
+        (let [job-ent (d/entity (d/db conn) job)]
+          (is (= 10 (:job/max-retries job-ent)))
+          (is (= :job.state/running (:job/state job-ent))))))
+
+    (testing "same retries on completed jobs with attempts remaining"
+      (let [[job _] (create-dummy-job-with-instances conn
+                                                     :job-state :job.state/completed
+                                                     :retry-count 2
+                                                     :instances [{:instance-status :instance.status/failed}])
+            job-uuid (:job/uuid (d/entity (d/db conn) job))]
+        (util/retry-job! conn job-uuid 2)
+        (let [job-ent (d/entity (d/db conn) job)]
+          (is (= 2 (:job/max-retries job-ent)))
+          (is (= :job.state/waiting (:job/state job-ent))))))
+
+    (testing "same retries on completed jobs with mea-culpa attempts remaining"
+      (let [[job _] (create-dummy-job-with-instances conn
+                                                    :job-state :job.state/completed
+                                                    :retry-count 1
+                                                    :instances [{:instance-status :instance.status/failed
+                                                                 :reason :preempted-by-rebalancer}])
+            job-uuid (:job/uuid (d/entity (d/db conn) job))]
+        (util/retry-job! conn job-uuid 1)
+        (let [job-ent (d/entity (d/db conn) job)]
+          (is (= 1 (:job/max-retries job-ent)))
+          (is (= :job.state/waiting (:job/state job-ent))))))
+
+    (testing "same retries on job with no attempts remaining"
+      (let [[job _] (create-dummy-job-with-instances conn
+                                                     :job-state :job.state/completed
+                                                     :retry-count 1
+                                                     :instances [{:instance-status :instance.status/failed}])
+            job-uuid (:job/uuid (d/entity (d/db conn) job))]
+        (util/retry-job! conn job-uuid 1)
```

Comment:
Should this throw an error?

Reply:
Discussed offline, keeping the behavior for compatibility.